### PR TITLE
Add unit test for BindingSourceDesigner.cs file

### PR DESCRIPTION
**Pull request by Olina-Zhang**
Related https://github.com/dotnet/winforms/issues/10773

## Proposed changes

-  Add unit test file: BindingSourceDesignerTests.cs for public property and method of the BindingSourceDesigner.cs file.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11775)

### CODE REVIEW

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/BindingSourceDesignerTests.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+using System.Reflection;
+
+namespace System.Windows.Forms.Design.Tests;
+
+public class BindingSourceDesignerTests
+{
+    [Fact]
+    public void BindingUpdatedByUser_SetValue_ShouldUpdateField()
+    {
+        BindingSourceDesigner designer = new()
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/BindingSourceDesignerTests.cs:18]: Please make sure this is disposed by added a `using`

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/BindingSourceDesignerTests.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        {
+            BindingUpdatedByUser = true
+        };
+
+        FieldInfo? fieldInfo = designer.GetType().GetField("_bindingUpdatedByUser", BindingFlags.NonPublic | BindingFlags.Instance);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/BindingSourceDesignerTests.cs:23]: Is it possible to use TestAccessor here?

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/BindingSourceDesignerTests.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        Mock<IServiceProvider> serviceProviderMock = new();
+        serviceProviderMock.Setup(sp => sp.GetService(typeof(IComponentChangeService))).Returns(componentChangeServiceMock.Object);
+
+        Mock<IComponent> componentMock = new();
+        Mock<ISite> siteMock = new();
+        siteMock.Setup(s => s.GetService(typeof(IComponentChangeService))).Returns(componentChangeServiceMock.Object);
+        componentMock.Setup(c => c.Site).Returns(siteMock.Object);
+
+        BindingSource bindingSource = new() { DataSource = componentMock.Object };
+        BindingSourceDesigner designer = new();
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/BindingSourceDesignerTests.cs:45]:
```suggestion
        using BindingSourceDesigner designer = new();
```

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/BindingSourceDesignerTests.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        Mock<IComponentChangeService> componentChangeServiceMock = new();
+        Mock<IServiceProvider> serviceProviderMock = new();
+        serviceProviderMock.Setup(sp => sp.GetService(typeof(IComponentChangeService))).Returns(componentChangeServiceMock.Object);
+
+        Mock<IComponent> componentMock = new();
+        Mock<ISite> siteMock = new();
+        siteMock.Setup(s => s.GetService(typeof(IComponentChangeService))).Returns(componentChangeServiceMock.Object);
+        componentMock.Setup(c => c.Site).Returns(siteMock.Object);
+
+        BindingSource bindingSource = new() { DataSource = componentMock.Object };
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/BindingSourceDesignerTests.cs:44]:
```suggestion
        using BindingSource bindingSource = new() { DataSource = componentMock.Object };
```

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/BindingSourceDesignerTests.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        BindingSourceDesigner designer = new();
+        designer.Initialize(bindingSource);
+
+        ComponentEventArgs args = new(componentMock.Object);
+
+        MethodInfo? onComponentRemovingMethod = typeof(BindingSourceDesigner).GetMethod("OnComponentRemoving", BindingFlags.NonPublic | BindingFlags.Instance);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/BindingSourceDesignerTests.cs:50]: I think TestAccessor can help here

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/BindingSourceDesignerTests.cs (hunk 2 above)
Comment (Olina-Zhang) [reply src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/BindingSourceDesignerTests.cs:23]: ITestAccessor cannot be found.

Comment (LeafShi1) [reply src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/BindingSourceDesignerTests.cs:23]: The BindingUpdatedByUser only a set property, maybe the test should be as following

```
using BindingSourceDesigner designer = new();

bool originalValue = designer.TestAccessor().Dynamic._bindingUpdatedByUser;
originalValue.Should().BeFalse();

designer.BindingUpdatedByUser = true;
bool updatedValue = designer.TestAccessor().Dynamic._bindingUpdatedByUser;
updatedValue.Should().BeTrue();
```

File: src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/BindingSourceDesignerTests.cs (hunk 5 above)
Comment (LeafShi1) [reply src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/BindingSourceDesignerTests.cs:50]:
```
ComponentEventArgs args = new(componentMock.Object);
designer.TestAccessor().Dynamic.OnComponentRemoving(null, args);
```

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**